RemovedInPytest4Warning: MarkInfo objects are deprecated

As indirectly reported in #8134 (see https://gist.github.com/aphearin/ae979da4dab3dbabf23929df1307b6d2).

Example usage:
```python
def value_from_markers(key, request):
    try:
        val = request.keywords[key].args[0]
    except KeyError:
        val = DEFAULTS[key]
    return val

@pytest.fixture
def ccd_data(request):
    """
    Return a CCDData object with units of ADU.

    The size of the data array is 100x100 but can be changed using the marker
    @pytest.mark.data_size(N) on the test function, where N should be the
    desired dimension.

    Data values are initialized to random numbers drawn from a normal
    distribution with mean of 0 and scale 1.

    The scale can be changed with the marker @pytest.marker.scale(s) on the
    test function, where s is the desired scale.

    The mean can be changed with the marker @pytest.marker.scale(m) on the
    test function, where m is the desired mean.
    """
    size = value_from_markers('data_size', request)
    scale = value_from_markers('data_scale', request)
    mean = value_from_markers('data_mean', request)
```

Resulting error:
```
self = <MarkInfo Mark(name='data_size', args=(10,), kwargs={})>

    def warned(self):
>       warnings.warn(warning, stacklevel=2)
E       _pytest.warning_types.RemovedInPytest4Warning: MarkInfo objects are deprecated as they contain merged marks which are hard to deal with correctly.
E       Please use node.get_closest_marker(name) or node.iter_markers(name).
E       Docs: https://docs.pytest.org/en/latest/mark.html#updating-code

.../_pytest/mark/structures.py:25: RemovedInPytest4Warning
```

--- a/astropy/nddata/tests/test_ccddata.py
+++ b/astropy/nddata/tests/test_ccddata.py
@@ -34,11 +34,11 @@ DEFAULT_DATA_SCALE = 1.0
 
 
 def value_from_markers(key, request):
-    try:
-        val = request.keywords[key].args[0]
-    except KeyError:
-        val = DEFAULTS[key]
-    return val
+    m = request.node.get_closest_marker(key)
+    if m is not None:
+        return m.args[0]
+    else:
+        return DEFAULTS[key]
 
 
 @pytest.fixture

